scripts/smart_contracts/check_contract.py

```diff
@@ -0,0 +1,26 @@
+from typer import Argument, run
+
+from vitrina.smart_contracts.exceptions import InvalidAdocError
+from vitrina.smart_contracts.services import has_valid_signature, is_checksum_valid
+
+
+def main(
+    adoc_file_path: str = Argument(..., help="Path to the ADOC file"),
+    expected_checksum: str = Argument(..., help="Expected SHA256 checksum for the PDF"),
+) -> None:
+    try:
+        if has_valid_signature(adoc_file_path):
+            print("Signature found")
+        else:
+            print("Signature not found")
+
+        if is_checksum_valid(adoc_file_path, expected_checksum):
+            print("Checksum matches")
+        else:
+            print("Checksum doesn't match")
+    except InvalidAdocError as e:
+        print(f"Invalid ADOC file: {e}")
```

Comment:
Not sure about practices in scripting here, but in CLI tools returning non-zero exit codes on error is standard practice:

`import sys`
`sys.exit(1)`